traceevents/src/main/java/com/tomtom/kotlin/traceevents/Tracer.kt

```diff
@@ -654,6 +665,44 @@ class Tracer private constructor(
                 e.message
             }
 
+        private fun getCaller(): String {
+
+            /**
+             * For Java 9, we suggest the following optimization:
+             *
+             * Use the [StackWalker] instead of [Thread.getStackTrace], because it's much, much
+             * faster to not get all frames.
+             *
+             * Get the stack up to a maximum of [STACK_TRACE_DEPTH] levels deep. Our caller must be
+             * in those first calls. If it's not, the unit tests fails and informs the developer
+             * the reconsider the value of [STACK_TRACE_DEPTH].
+             *
+             * <pre>
+             *     val stack = StackWalker.getInstance().walk {
+             *         frames -> frames.limit(STACK_TRACE_DEPTH).collect(Collectors.toList())
+             * </pre>
+             *
+             * For now, use the slower Java 8 version:
+             */
+            val stack = Throwable().stackTrace
+
+            // Find "com.sun.proxy.$Proxy" function on stack that called this function.
+            var i = 0
+            while (i < stack.size && !stack[i].className.startsWith("com.sun.proxy.\$Proxy")) i++
```

Comment:
As discussed, may be good to replace this check with one that doesn't rely on a `sun` implementation that we don't explicitly depend on ourselves. (I tried to find whether it's always this one or if it can differ per system environment but couldn't find much information.)

Reply:
Done